cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, err
 	}
+	maps.Copy(vars, envMap)
```

Comment:
**Code Structure Issue**: The use of 'maps.Copy' for copying environment variables and opts.Variables could lead to unintended overwrites if keys overlap. The order of copying could be crucial and needs clear documentation or reconsideration to prevent potential bugs. <br> **Fix**: Ensure that the order of copying variables is intentional and document the behavior clearly. If the order is not intended, consider merging the variables with a method that handles key conflicts explicitly. <br> **Code Suggestion**: 
 ```
 -	maps.Copy(vars, opts.Variables)
 +	// Ensure no key conflicts or handle them explicitly
 +	maps.Merge(vars, opts.Variables)
 +	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
 +	if err != nil && !errors.Is(err, os.ErrNotExist) {
 +		return nil, err
 +	}
 +	maps.Merge(vars, envMap)  # Handle potential key conflicts during merge
 ```